src/pool/cli.c

```diff
@@ -1974,6 +1981,7 @@ map_refresh(tse_task_t *task)
 
 	D_DEBUG(DB_MD, DF_UUID": %p: asking rank %u for version > %u\n",
 		DP_UUID(pool->dp_pool), task, rank, version);
+	dc_pool_put(pool);
```

Comment:
Can we hold the reference during the RPC process on server side, and then ask map_refresh_cb() to release the reference? Then map_refresh_cb() do not need to get reference again. Otherwise, if it is possible that the pool may be freed before map_refresh_cb() triggered, then  dc_pool_get() in map_refresh_cb() may access invalid memory?

Reply:
leave it to @liw reply as this part is backport from one of his PR, thx.

Reply:
@Nasf-Fan, `arg->mra_pool` is the reference that prevents the `dc_pool` from being freed before `map_refresh_cb` accesses it. The extra references taken in this PR are for the tricky `reinit` cases, where anther thread could execute the reinitialized task and (in theory) free the `arg->mra_pool` reference after the `reinit` call but before the function that calls `reinit` returns.